### ISSUE TITLE
Aqara Magic Cube - Inverted behavior

### ISSUE THREAD

**Issue by Dillton** [golles/Home-Assistant-Config#5]
### Blueprint filename

zigbee2mqtt_aqara_magic_cube.yaml

### The problem

Hello.

I was using your Z2M blueprint for Aqara magic cube for a very long time, the old one with just "_action" entity required.
Thank you for that it's simple and easy, exactly what i need in my use case.

Anyway I was forced to make some changes on Z2M which included channel switch and repairing all devices.
Unfortunatelly I think because of change in MQTT naming convention inside HomeAssistant after repairing the Aqara cube the old blueprint no longer works. I tried of course move to new blueprint with also "side" entity required but the actions defined seems at least from my point of view inverted.

What I mean i have defined simple flip actions which is basically all I need for simple light controlling.
For example flip cube to side 0 has defined action to turn off all lights in one room.
Flip cube to side 1 has action to turn on specific light.

Now what is actually happening. When I flip the cube from side 0 to side 1 nothing happens.
Then if I flip back from side 1 to 0 the light defined in action for flip to side 1 is turned on.
Step three if I again flip from side 0 to 1 the light is turned off.

What I think is that there is some problem with handling or reading states from variables correctly.
In other words the automation actually reacts when you are flipping "from side" not "to side".
Also I tried other action like rotate with side 1 up and that worked as expected so I am guessing only flip actions are affected.

I tried to figure this out by myself but I am not a template master so I would really appreciate your help.

### What version of Home Assistant Core has the issue?

2024.5.5

**Comment by Dillton** [comment]
Okay update.

I deep dived into that once again and found that "_side" entity is not updated fast enough
and when the automation starts declaring variables it still takes the old value for "side variable".

So I tried to move the variables declaration right before the flip choose action as shown below to get some more miliseconds of time for the entity to update but it didn't worked for me so I suggest it still was't fast enough.
  - conditions: '{{ event == "flip180" or event == "flip90" }}'
    sequence:
    - variables:
        side_entity: !input side
        side: '{{ states(side_entity) }}'
    - choose:
      - conditions: '{{ side == 0 }}'
        sequence: !input flip_side_0
      - conditions: '{{ side == 1 }}'
        sequence: !input flip_side_1
      - conditions: '{{ side == 2 }}'
        sequence: !input flip_side_2
      - conditions: '{{ side == 3 }}'
        sequence: !input flip_side_3
      - conditions: '{{ side == 4 }}'
        sequence: !input flip_side_4
      - conditions: '{{ side == 5 }}'
        sequence: !input flip_side_5

Finally what I found as working so far and hope it will still work in future was to add before variables declaration a simple wait for trigger like this:
- wait_for_trigger:
    - platform: state
      entity_id: !input side
  continue_on_timeout: false

Now everything is working again exactly as expected.

**Comment by golles** [comment]
Hi, thanks for raising this issue and proposing a potential fix!

I'm only using 1 of my 2 cubes and have it defined here: https://github.com/golles/Home-Assistant-Config/blob/main/automations/logeerkamer/cube.yaml
I'll need to double-check it that still works, as I'm not using that very often

**Comment by Dillton** [comment]
No problem and thanks for zour response
.
As I mentioned I have reworked whole Z2M network so the new pairing of the cube is probably a bit different than before
which could lead to some additional miliseconds for changing the side entity value.
